### PR TITLE
add makeTemplate and validateTemplate to plotly_js function reference.

### LINKED ISSUE #1111: Document the new plotly.js makeTemplate function and layout.template attr
This feature is pretty cool and needs some docs (or at least I couldn't find any!): https://github.com/plotly/plotly.js/blob/fc0a1f8313abc17d188481eaabd6dfc1a1fb0922/src/plot_api/template_api.js

### PR DESCRIPTION
**Pull request by michaelbabyn**
resolves #1111

### CODE REVIEW

**Comment by nicolaskruchten** [comment]
@VeraZab check this out.

File: _posts/reference_pages/2016-06-03-plotly_js_function_ref.html [hunk 1]
```diff
@@ -361,6 +361,77 @@ <h4 id="plotly-validate"><a href="{{ BASE_URL }}/javascript/plotlyjs-function-re
...
+<code>Plotly.makeTemplate</code> copies the style information from a figure. It does this by returning a <code>template</code> object which can be passed to the <code>layout.template</code> attribute of another figure.
+<br /><br />
+<fieldset class="signatures">
+<legend>Signature</legend>
+<dl>
+  <dt><code>Plotly.makeTemplate(figure)</code></dt>
+  <dd>
+    <dl>
+      <dt><code>figure</code></dt>
+      <dd>a plot object, with <code>{data, layout}</code> members
```
Comment (alexcjohnson) [inline _posts/reference_pages/2016-06-03-plotly_js_function_ref.html]: Or a `<div>` DOM element with a plot already in it - at least once v1.42 is out (next week) with https://github.com/plotly/plotly.js/pull/3111 (thanks @antoinerg!)

File: _posts/reference_pages/2016-06-03-plotly_js_function_ref.html [hunk 2]
```diff
@@ -361,6 +361,77 @@ <h4 id="plotly-validate"><a href="{{ BASE_URL }}/javascript/plotlyjs-function-re
...
+a problem, it depends on what you're using the template for.
+<br /><br />
+<fieldset class="signatures">
+<legend>Signature</legend>
+<dl>
+  <dt><code>Plotly.validateTemplate(figure, template)</code></dt>
+  <dd>
+    <dl>
+      <dt><code>figure</code></dt>
+      <dd>a plot object, with <code>{data, layout}</code> members
```
Comment (alexcjohnson) [inline _posts/reference_pages/2016-06-03-plotly_js_function_ref.html]: this too can be a DOM element (with or without the PR mentioned above)

**Comment by alexcjohnson** [comment]
Very nice @michaelbabyn!

I'm assuming there will be a separate PR to document the structure of a `template` better? We have https://plot.ly/javascript/reference/#layout-template but I think it needs a lot more detail, with some examples, to really be useful.

**Pull request merged**